docs/realtime.rst

```diff
@@ -0,0 +1,117 @@
+The Hypothesis Real Time API
+============================
+
+.. warning::
+
+   This document describes an API that is in the early stages of being
+   documented and refined for public use. Details may change, and your systems
+   may break in the future.
+
+In addition to the `HTTP API <http://h.readthedocs.io/en/latest/api/>`_
+Hypothesis has a WebSocket-based API that allows developers to receive near
+real-time notifications of annotation events.
+
+Overview
+--------
+
+.. note::
+
+   At the moment, use of the Real Time API is limited to server-side clients.
+   JavaScript applications running in the browser **will not be able to use this
+   API** due to security restrictions. We hope to lift this restriction in the
+   future.
+
+To use the Real Time API, you should open a WebSocket connection to the
+following endpoint::
+
+    wss://hypothes.is/ws
```

Comment:
One thing worth thinking about is that this is actually a pretty implausible way of managing the websocket in the long term, because it requires all these long-lived connections to go through this one DNS address.

Slack works around this problem by requiring users to make an HTTP API call which returns a "capability" endpoint URL. This gives a couple of nice properties:

- WebSockets don't need any authentication at all -- any auth is provided by the capability URL.
- Sharding WS connections across domain names is easy.
- The API call can be responsible for any necessary rate-limiting or black/whitelisting, thus simplifying the WebSocket server.